OsmAnd/src/net/osmand/plus/importfiles/tasks/SaveGpxAsyncTask.java

```diff
@@ -92,6 +93,7 @@ protected String doInBackground(Void... nothing) {
 				}
 				GpxDataItem item = new GpxDataItem(resultFile, gpxFile);
 				app.getGpxDbHelper().add(item);
+				app.getSmartFolderHelper().addTrackItemToSmartFolder(new TrackItem(resultFile));
```

Comment:
We have several more places where gpx are saved to app tracks dir, for example SaveCurrentTrackTask, SaveDirectionsAsyncTask, CoordinateInputDialogFragment.SaveGpxAsyncTask etc. 

Probably it is not a big issue, if trackItems for smart folders are updated after user gets to tracks screens